DeusEx/Classes/DXREntranceRando.uc

```diff
@@ -237,37 +201,122 @@ function bool ValidateConnections()
             //If map can be visited, go through all the places it can go
             if (canVisitMap)
             {
-                for (k=0;k<mapdests[j].numDests;k++)
-                {
-                    //If any of those destinations are not in the places we can visit, add them
-                    canVisitDest = False;
-                    for (p=0;p<visitable;p++)
-                    {
-                        if ( canvisit[p]==mapdests[j].dest[k] )
-                        {
-                            canVisitDest = True;
-                        }
-                    }
-                    if (canVisitDest == False)
-                    {
-                        canvisit[visitable] = mapdests[j].dest[k];
-                        visitable++;
-                    }
-                    
-
-                }
-            
+                MarkMapDestinationsVisitible(mapdests[j], canvisit, visitable);
             }
         }
     }
     
     //Theoretically I should probably actually check to see if the maps match,
```

Comment:
what do you mean by this comment?

Reply:
The thought behind it was that I'm collecting a list (and count) of the maps that I can visit based on the teleporters, but I'm just checking to see if the number of maps I can visit matches the the number of maps in the mission.  It would probably be more safe to actually verify that the list of visitable maps actually matches the list of maps in the mission, which could catch a scenario where maybe a single map gets added to the list twice, or we somehow visit a map that we shouldn't be able to.

Since the above two scenarios are fairly unlikely (once the code has been tested sufficiently), just checking the number of maps in each list is basically an equivalent check (But not quite as rigorous)